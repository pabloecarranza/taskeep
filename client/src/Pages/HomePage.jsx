import React from 'react';
import { Sidebar } from './../components/Sidebar';
import { TaskArea } from './../components/TaskArea';
import { Box, Flex } from '@chakra-ui/react';

export const HomePage = () => {
	return (
		<Flex justifyContent='center' mt='5px'>
			<Sidebar />
			<TaskArea />
		</Flex>
	);
};
