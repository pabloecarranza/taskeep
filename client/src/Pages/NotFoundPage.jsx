import {
	Heading,
	Container,
	Box,
	Center,
	Button,
	Text,
	Image,
} from '@chakra-ui/react';
import Logo from '../assets/daily-tasks.png';
import { ArrowForwardIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

export const NotFoundPage = ({
	error_text,
	message_redirect,
	page_redirect,
	test = false,
}) => {
	let navigate = useNavigate();
	const reDirectTo = () => {
		navigate('/homepage', { replace: true });
	};

	return (
		<Container centerContent>
			<Center padding='4' w='100vw' h='100vh'>
				<Box
					w='80%'
					h='80%'
					display='flex'
					flexDir='column'
					alignItems='center'
					justifyContent='center'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Heading as='h2' size='2xl' fontWeight='bold' textAlign='center'>
						{error_text}
						<Text
							bgGradient={test ? '' : 'linear(to-l, #7928ca, #0084ff)'}
							bgClip='text'
							fontSize='3xl'
							fontWeight='bold'
							pb='30px'
							pt='20px'
						>
							{message_redirect}
						</Text>
					</Heading>
					<Box>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							size='lg'
							onClick={reDirectTo}
						>
							{page_redirect}
						</Button>
					</Box>
				</Box>
			</Center>
		</Container>
	);
};

NotFoundPage.propTypes = {
	error_text: PropTypes.string.isRequired,
	message_redirect: PropTypes.string.isRequired,
	page_redirect: PropTypes.string.isRequired,
};
