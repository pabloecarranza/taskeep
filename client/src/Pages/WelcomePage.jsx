import {
	Box,
	Flex,
	Text,
	Heading,
	Button,
	Center,
	Image,
	useDisclosure,
	ScaleFade,
	Link,
} from '@chakra-ui/react';
import { ExternalLinkIcon } from '@chakra-ui/icons';
import { BsLinkedin, VscGithubInverted } from 'react-icons/all';
import Logo from '../assets/daily-tasks.png';
import Typewriter from 'typewriter-effect';
import { LoginForm } from '../components/LoginForm';
import { ArrowForwardIcon } from '@chakra-ui/icons';

import bg from '../assets/hojas.png';

export const WelcomePage = () => {
	const { isOpen, onToggle } = useDisclosure();

	return (
		<Flex
			w='100vw'
			h='100vh'
			flexDir={{ base: 'column', md: 'row', lg: 'row' }}
		>
			<Box w='100%' h='100%'>
				<Center
					w='100%'
					h='100%'
					flexDir='column'
					alignItems='flex-end'
					textAlign='right'
					pr='35px'
				>
					<Box>
						<Image src={Logo} alt='Logo_Task' boxSize='150px' />
					</Box>
					<Text
						bgGradient='linear(to-l, #b785e9, #61acf3)'
						bgClip='text'
						fontSize='6xl'
						fontWeight='extrabold'
						pb='10px'
					>
						TASKEEP
					</Text>
					<Heading as='h2' size='2xl' noOflines={1} fontWeight='bold'>
						<Typewriter
							options={{
								strings: ['The most powerful', 'The real powerful'],
								autoStart: true,
								loop: true,
							}}
						/>

						<Text
							bgGradient='linear(to-l, #7928ca, #0084ff)'
							bgClip='text'
							fontSize='5xl'
							fontWeight='bold'
							pb='50px'
						>
							Task Management App
						</Text>
					</Heading>
					<Box>
						<Button
							size='lg'
							colorScheme='blue'
							rightIcon={<VscGithubInverted />}
							mr='10px'
							onClick={() =>
								window.open(
									'https://github.com/pabloecarranza/taskeep',
									'_blank'
								)
							}
						>
							GitHub
						</Button>
						<Button
							rightIcon={<ArrowForwardIcon />}
							colorScheme='blue'
							variant='outline'
							size='lg'
							onClick={onToggle}
						>
							Get Started
						</Button>
					</Box>

					<Button
						colorScheme='gray'
						variant='link'
						rightIcon={<BsLinkedin />}
						pt='35px'
						onClick={() =>
							window.open(
								'https://www.linkedin.com/in/pabloecarranza/',
								'_blank'
							)
						}
					>
						Made by Pablo Carranza
					</Button>
				</Center>
			</Box>
			<Center
				w='80%'
				h='100%'
				backgroundImage={bg}
				backgroundPosition='center'
				backgroundRepeat='no-repeat'
				backgroundSize='cover'
			>
				<ScaleFade initialScale={0.9} in={isOpen}>
					<LoginForm />
				</ScaleFade>
			</Center>
		</Flex>
	);
};
