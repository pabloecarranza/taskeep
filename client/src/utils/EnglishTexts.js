export const NotFoundPageDates = {
	error_text: "We're sorry - something's gone wrong.",
	message_redirect:
		'Our team has been notified, but click below to go HomePage',
	page_redirect: 'HomePage',
};

export const WelcomePagesDates = {
	app_name: 'TASKEEP',
	slogan_one: 'The most powerful',
	slogan_two: 'The real powerful',
	subtitle: 'Task Management App',
	repository: 'Github',
	start_app: 'Get Starter',
	made: 'Made by Pablo Carranza',
	test: false,
};

export const ModalWelcomeDates = {
	greeting: 'Hi',
	welcome_text_1: 'Welcome to my personal app for task management.',
	welcome_text_2:
		'The main functionalities are to create tasks and group them by lists, set the priority and expiration date. I hope you enjoy it !',
	name: 'Pablo',
	button_text: 'Close',
};

export const ModalConfirmDates = {
	title_partOne: 'Delete ',
	title_partTwo: 'task list',
	message:
		"Are you sure? You can't undo this action afterwards and all task assigned to this list be eliminated",
	button_one: 'Cancel',
	button_two: 'Delete',
};

export const ModalAddListDates = {
	title: 'Create a new task list',
	name: 'List name',
	button_one: 'Cancel',
	button_two: 'Save',
	place_holder: 'Write here...',
};
