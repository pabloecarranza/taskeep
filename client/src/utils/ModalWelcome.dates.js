export const ModalWelcomeDates = {
	greeting: 'Hi',
	welcome_text_1: 'Welcome to my personal app for task management.',
	welcome_text_2:
		'The main functionalities are to create tasks and group them by lists, set the priority and expiration date. I hope you enjoy it !',
	name: 'Pablo',
};
