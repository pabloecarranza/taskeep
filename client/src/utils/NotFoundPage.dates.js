export const NotFoundPageDates = {
	error_text: "We're sorry - something's gone wrong.",
	message_redirect:
		'Our team has been notified, but click below to go HomePage',
	page_redirect: 'HomePage',
};
