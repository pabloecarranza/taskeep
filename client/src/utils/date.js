export function fecha() {
	const meses = [
		'January',
		'February',
		'March',
		'April',
		'May',
		'June',
		'July',
		'August',
		'September',
		'Octuber',
		'November',
		'December',
	];

	const dias_semana = [
		'Sunday',
		'Monday',
		'Tuesday',
		'Wednesday',
		'Thursday',
		'Friday',
		'Saturday',
	];

	const fecha = new Date();

	let date =
		dias_semana[fecha.getDay()] +
		', ' +
		fecha.getDate() +
		' de ' +
		meses[fecha.getMonth()] +
		' de ' +
		fecha.getUTCFullYear();

	return date;
}

export function validate(input) {
	let errors = {};
	const regexName = /^([a-zA-Z ]+)$/i;
	input.name
		? (errors.name = '')
		: (errors.name = 'El nombre no puede estar vacio');
	if (input.name && !regexName.test(input.name)) {
		errors.name = 'El nombre no puede incluir caracteres especiales o numeros';
	}

	return errors;
}

export function diaDeSemana() {
	const dias_semana = [
		'Sunday',
		'Monday',
		'Tuesday',
		'Wednesday',
		'Thursday',
		'Friday',
		'Saturday',
	];

	const fecha = new Date();

	let date = dias_semana[fecha.getDay()];

	return date;
}

export function MesyDia() {
	const meses = [
		'January',
		'February',
		'March',
		'April',
		'May',
		'June',
		'July',
		'August',
		'September',
		'Octuber',
		'November',
		'December',
	];

	const fecha = new Date();

	let date = fecha.getDate() + ' de ' + meses[fecha.getMonth()];
	return date;
}
