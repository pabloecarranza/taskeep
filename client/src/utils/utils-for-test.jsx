import { render } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { authSlice } from '../features/api/authSlice';
import { listSlice } from '../features/api/listSlice';
import { sessionSlice } from '../features/api/sessionSlice';
import { taskSlice } from '../features/api/taskSlice';

export function renderWithProviders(
	ui,
	{
		preloadedState = {},
		store = configureStore({
			reducer: {
				authSlice: authSlice.reducer,
				listSlice: listSlice.reducer,
				taskSlice: taskSlice.reducer,
				session: sessionSlice.reducer,
			},
			middleware: getDefaultMiddleware =>
				getDefaultMiddleware().concat(
					authSlice.middleware,
					listSlice.middleware,
					taskSlice.middleware
				),
			preloadedState,
		}),
		...renderOptions
	} = {}
) {
	// eslint-disable-next-line react/prop-types
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}

	// Return an object with the store and all of RTL's query functions
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
