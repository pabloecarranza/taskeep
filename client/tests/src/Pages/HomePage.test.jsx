import { render, screen } from '@testing-library/react';
import { vi, expect, describe, it, test } from 'vitest';
import { BrowserRouter as Router } from 'react-router-dom';
import { HomePage } from './../../../src/Pages/HomePage';
import { renderWithProviders } from './../../../src/utils/utils-for-test';

describe('Test suite on HomePage component', () => {
	vi.mock('../../../src/components/Sidebar/Sidebar', () => {
		return {
			__esModule: true,
			default: () => {
				return <Sidebar />;
			},
			Sidebar: () => {
				return <div />;
			},
		};
	});

	vi.mock('../../../src/components/Tabs/MainArea', () => {
		return {
			__esModule: true,
			default: () => {
				return <MainArea />;
			},
			MainArea: () => {
				return <div />;
			},
		};
	});

	it('should ', () => {
		const { container } = render(<HomePage />);
		expect(container).toMatchSnapshot();
	});
});
