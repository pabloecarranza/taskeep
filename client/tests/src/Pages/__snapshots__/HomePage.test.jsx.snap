// Vitest Snapshot v1

exports[`Test suite on HomePage component > should  1`] = `
<div>
  <div
    class="css-ctfmb8"
  >
    <div />
    <div />
  </div>
</div>
`;
